tests/test_dictdiffer.py

```diff
@@ -54,6 +54,14 @@ def test_change(self):
         diffed = next(diff(first, second))
         assert ('change', 'a', ('b', None)) == diffed
 
+        first = {'a': 10.0}
+        second = {'a': 10.5}
+        diffed = list(diff(first, second, tolerance=0.1))
+        assert [] == diffed
+
+        diffed = next(diff(first, second, tolerance=0.01))
+        assert ('change', 'a', (10.0, 10.5)) == diffed
+
```

Comment:
For sure I could try to cast string to int or float, but is it really what you want ?
I think the default behavior should be to assume that the types of the values being compared are coherent. Maybe a flag 'allow_cast' can be added to allow the comparison of '10' and 10.5.